### ISSUE TITLE
how to set the seed for mimc hash

### ISSUE THREAD

**Issue by mdj33** [ConsenSys/gnark#55]
hi, could you tell me how to select the seed for mimc hash?  what principle should be followed or noticed?  thanks!

```
hash.MIMC_BN256.New("seed1")
```

**Comment by ThomasPiellard** [comment]
Hi, as it is now the seed should be a random string, without particular principles about it. The constants used in the mimc `encrypt` function are derived from computing successive `SHA3` of the seed. 

This part is a bit dirty, and we will change it at some point so `hash.MIMC_BN256.New` takes a `io.Reader` as source of randomness (like what you would find in golang's crypto/ package). We implemented the `New` function for the hashes like this because when using mimc in a snark circuit, we need to instantiate the mimc gadget with the same seed as the one used in the real go implementation.

We did a clean pass on eddsa to fix this kind of problem, and we will do the same with the hash functions.
